src/net_processing.cpp

```diff
@@ -3687,6 +3674,25 @@ bool PeerLogicValidation::ProcessMessages(CNode* pfrom, std::atomic<bool>& inter
     return fMoreWork;
 }
 
+bool PeerLogicValidation::ProcessGlobalTasks(std::atomic<bool>& interruptMsgProc)
+{
+    // Reconsider orphan transactions that may now have parents
+    {
+        LOCK2(cs_main, g_cs_orphans);
+        if (!g_orphan_work_set.empty()) {
```

Comment:
That's exactly how I'd normally structure a function like this. However, I expect we'll want to move more housekeeping tasks into `ProcessGlobalTasks()`, in which case we'll want to continue through to the next task queue if there are not orphan txs to reconsider. I didn't use the early exit pattern so that adding those tasks doesn't require restructuring this function.